### PR TITLE
testing that a new protocol is correctly tied to a service

### LINKED ISSUE #235: CreateProtocol needs to respect ServiceID
Now: if a protocol that has been started by a service creates a new protocol, this new protocol will not be attached to the service and so it will not call the `NewProtocol` method of the service, making it impossible for the service to set configuration options of this protocol.

Wanted: The `CreateProtocol` method of `TreeNodeInstance` should copy the `ServiceID` so that the new protocol is also attached to the service.

### PR DESCRIPTION
**Pull request by ineiti**
Closes #235

### CODE REVIEW

File: treenode.go [hunk 1]
```diff
@@ -673,7 +673,7 @@ func (n *TreeNodeInstance) SendToChildrenInParallel(msg interface{}) error {
 // returns it with any error that might have happened during the creation. This
 // protocol is only handled by onet, no service are "attached" to it.
 func (n *TreeNodeInstance) CreateProtocol(name string, t *Tree) (ProtocolInstance, error) {
-	pi, err := n.overlay.CreateProtocol(name, t, NilServiceID)
+	pi, err := n.overlay.CreateProtocol(name, t, n.Token().ServiceID)
```
Comment (nikkolasg) [inline treenode.go]: Lol. That was easy ;)

**Comment by nikkolasg** [comment]
LGTM

File: treenode.go [hunk 2]
```diff
@@ -673,7 +673,7 @@ func (n *TreeNodeInstance) SendToChildrenInParallel(msg interface{}) error {
 // returns it with any error that might have happened during the creation. This
```
Comment (jeffallen) [inline treenode.go]: Please update comment to new behavior, also s/instantiates/instantiate/

File: treenode.go (hunk 1 above)
Comment (ineiti) [reply treenode.go]: First I wanted to submit only this line - then I thought that I have 2 engineers that will tell me to write some test...

**Pull request merged**